src/password_manager.rs

```diff
@@ -0,0 +1,244 @@
+use cosmic::{iced::Length, widget, Element};
+
+use crate::{fl, Message};
+
+pub struct PasswordManager {
+    pub input_description: String,
+    pub input_password: String,
+    pub cached_password_list: Option<Vec<String>>,
+}
+
+impl PasswordManager {
+    pub fn new() -> Self {
+        Self {
+            input_description: Default::default(),
+            input_password: Default::default(),
+            cached_password_list: None,
+        }
+    }
+
+    pub fn populate_password_cache(&mut self) {
+        if self.cached_password_list.is_none() {
+            self.cached_password_list = Some(self.get_password_list());
+        }
+    }
+
+    pub fn clear_password_cache(&mut self) {
+        self.cached_password_list = None;
+    }
+
+    pub fn get_password(&self, identifier: &str) -> Option<String> {
+        let ss = match secret_service::blocking::SecretService::connect(
```

Comment:
This should be spawned with a `Command` with the async version so that it doesn't block the UI.

Reply:
Ok, as I said I'm new to iced, so this is a bit unfamiliar territory for me. But I will look in to this.

Reply:
You can return a `Command` from the `update` method of the application. We have a convenience method for creating commands that spawn futures, which is `cosmic::command::future()`.